Cython/Compiler/ModuleNode.py

```diff
@@ -2035,37 +2078,111 @@ def generate_richcmp_function(self, scope, code):
             # need to call up into base classes as we may not know all implemented comparison methods
             extern_parent = cls if cls.typeptr_cname else scope.parent_type.base_type
 
-        eq_entry = None
-        has_ne = False
+        total_ordering = scope.directives.get('total_ordering', False)
+
+        comp_entry = {}
+
         for cmp_method in TypeSlots.richcmp_special_methods:
             for class_scope in class_scopes:
                 entry = class_scope.lookup_here(cmp_method)
                 if entry is not None:
+                    comp_entry[cmp_method] = entry
                     break
+
+        if total_ordering:
+            # Check this is valid - we must have at least 1 operation defined.
+            comp_names = [from_name for from_name, to_name in TOTAL_ORDERING if from_name in comp_entry]
+            if not comp_names:
+                if '__eq__' not in comp_entry:
+                    error(scope.parent_type.pos,
+                          "total_ordering directive used, but no comparison and equality methods defined")
```

Comment:
Looks like I said before that this should not be an error. Let's not make it an error. :)
```suggestion
                    warning(scope.parent_type.pos,
                            "total_ordering directive used, but no comparison and equality methods defined")
```